### ISSUE TITLE
Add feature of installing other (.py for example) scripts to bin

### ISSUE THREAD

**Issue by AlexKaravaev** [ros2-rust/cargo-ament-build#1]
I think it will be quite useful feature since this way we will be able to have both python and rust nodes in one package. Similar functionality can be done with following CMakeLists for example:

```
add_executable(cpp_node/cpp_node.cpp)
ament_target_dependencies(cpp_node rclcpp tf2_ros)

ament_python_install_package(${PROJECT_NAME})
install(PROGRAMS
  py_node/py_node.py
  DESTINATION lib/${PROJECT_NAME}
)
```

I am not so familiar with cargo though, and I am not sure if that is possible. Well, to my understanding, python scripts are definetely possible, but don't know how to do it with cpp

Or

I think from it can be more structured to have only CMakeLists be capable of building all client libs packages(python, c++, rust) and cargo would be only for rust nodes. That way it's possible to just create some custom CMakeLists command like

```
ament_cargo_install_package
```

**Comment by nnmm** [comment]
Hi @AlexKaravaev, apologies, I only now saw this issue.

So, you already can install things to `share` (see the README), but you're right that `lib` and other places are not yet supported.

I will definitely also add an option to install to `lib` and `include`, maybe `bin` if there's a use case. However, I want you to also be able to install a file _from the build space_. To achieve that goal, I think we can have special variables to indicate cargo's TARGET_DIR or OUT_DIR:

[package.metadata.ros]
install_to_include = ["$OUT_DIR/include"]

**Comment by AlexKaravaev** [comment]
@nnmm Great!! With installing to lib it will be definitely easier to install python scripts. 

Do you think in the future it will be necessary to create ament plugin for cargo to run ?
```
ament_cargo_install_package
```

Just from a user perspective, when I would want to have both c++ and rust nodes in one package. I guess it's impossible with cargo to compile both at the same time right?

**Comment by nnmm** [comment]
@AlexKaravaev Correct, but you could simply call `cargo ament-build` from `CMakeLists.txt`. That installs the package in a ROS 2 layout.
